sdk/core/Azure.Core/src/ResponseError.Serialization.cs

```diff
@@ -0,0 +1,192 @@
+
+            public override void Write(Utf8JsonWriter writer, ResponseError? value, JsonSerializerOptions options)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        /// Writes the <see cref="ResponseError"/> to the provided <see cref="Utf8JsonWriter"/>.
+        /// </summary>
+        public void Write(Utf8JsonWriter writer, ModelReaderWriterOptions options)
+        {
+            writer.WriteStartObject();
+
+            if (Code != null)
+            {
+                writer.WritePropertyName("code");
+                writer.WriteStringValue(Code);
+            }
+
+            if (Message != null)
+            {
+                writer.WritePropertyName("message");
+                writer.WriteStringValue(Message);
+            }
+
+            if (Target != null)
+            {
+                writer.WritePropertyName("target");
+                writer.WriteStringValue(Target);
+            }
+
+            if (InnerError != null)
+            {
+                writer.WritePropertyName("innererror");
+                InnerError.Write(writer, ModelReaderWriterOptions.Json);
+            }
+
+            if (Details.Count > 0)
+            {
+                writer.WritePropertyName("details");
+                writer.WriteStartArray();
+
+                foreach (var detail in Details)
+                {
+                    if (detail == null)
+                    {
+                        writer.WriteNullValue();
+                    }
+                    else
+                    {
+                        detail.Write(writer, options);
+                    }
+                }
+
+                writer.WriteEndArray();
+            }
+
+            writer.WriteEndObject();
+        }
+
+        /// <inheritdoc />
+        public ResponseError Create(ref Utf8JsonReader reader, ModelReaderWriterOptions options)
+        {
+            var format = options.Format == "W" ? GetFormatFromOptions(options) : options.Format;
+            if (format != "J")
+            {
+                throw new FormatException($"The model {nameof(ResponseError)} does not support '{format}' format.");
+            }
+
+            using var document = JsonDocument.ParseValue(ref reader);
+            var element = document.RootElement;
+            return ReadFromJson(element) ?? new ResponseError();
+        }
+
+        /// <summary>
+        /// Writes the <see cref="ResponseError"/> to the provided <see cref="Utf8JsonWriter"/>.
+        /// </summary>
+        public BinaryData Write(ModelReaderWriterOptions options)
+        {
+            var format = options.Format == "W" ? GetFormatFromOptions(options) : options.Format;
+            if (format != "J")
+            {
+                throw new FormatException($"The model {nameof(ResponseError)} does not support '{format}' format.");
+            }
+
+            using var stream = new System.IO.MemoryStream();
```

Comment:
Should call into ModelReaderWriter here it uses a faster buffer than memory stream.  You can use RehydrationToken as a guide https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/RehydrationToken.Serialization.cs#L136-L147